### PR TITLE
ci: add NodeJS 20 to CI matrix

### PR DESCRIPTION
**Pull request by rbnayax**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?
Test are only running on node 16. Unit tests also on node 18

Issue Number: #12215

## What is the new behavior?
unit tests and integration tests will run on node 16, 18 and 20

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

**Comment by rbnayax** [comment]
@kamilmysliwiec can you please approve running the workflows? I can't be sure my changes work otherwise :-)

File: .circleci/config.yml [hunk 1]
```diff
@@ -130,7 +136,7 @@ jobs:
   codechecks_benchmarks:
     working_directory: ~/nest
     docker:
-      - image: cimg/node:16.20
+      - image: cimg/node:<< pipeline.parameters.maintenance-node-version >>
```
Comment (rbnayax) [inline .circleci/config.yml:139]: @kamilmysliwiec I wonder if we need to run this step also on other node version?

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,9 +1,25 @@
 version: 2
 
+parameters:
+  maintenance-node-version:
+    type: string
+    default: "16.20"
```
Comment (kamilmysliwiec) [inline .circleci/config.yml:6]: These may quickly get out of date - we will keep supporting v16 even after it's no longer a maintenance version (after September this year). That's why versions were hardcoded before

Comment (rbnayax) [reply .circleci/config.yml:9]: Ok, I handled it - added an optional "legacy" node version, but once node 16 becomes legacy, the "main" steps will run on the maintenance version which will be 18. Does that make sense?

**Comment by kamilmysliwiec** [comment]
LGTM

**Comment by kamilmysliwiec** [comment]
I'll investigate, thank you @rbnayax

**Comment by rbnayax** [comment]
Apparently you can validate the config locally - https://support.circleci.com/hc/en-us/articles/360006735753-How-to-validate-your-CircleCI-configuration
And I can reproduce the bug and managed to find a fix :-) Creating PR now

**Comment by rbnayax** [comment]
@kamilmysliwiec fix is in #12243

**Pull request merged**